controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -94,8 +96,71 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 func (r *Reconciler) deleteSpaceBinding(logger logr.Logger, spaceBinding *toolchainv1alpha1.SpaceBinding) error {
 	logger.Info("deleting the SpaceBinding")
+
+	// check if spaceBinding was created from SpaceBindingRequest,
+	// in that case we must delete the SBR and will take care of deleting the SpaceBinding as well
+	spaceBindingReqeuestAssociated := checkSpaceBindingRequestAssociated(spaceBinding)
+	if spaceBindingReqeuestAssociated.found {
+		return r.deleteSpaceBindingRequest(logger, spaceBindingReqeuestAssociated)
+	}
+
+	// otherwise delete the SpaceBinding resource directly ...
+	return r.DeleteSpaceBinding(spaceBinding)
+}
+
+func (r *Reconciler) DeleteSpaceBinding(spaceBinding *toolchainv1alpha1.SpaceBinding) error {
 	if err := r.Delete(context.TODO(), spaceBinding); err != nil {
 		return errs.Wrapf(err, "unable to delete the SpaceBinding")
 	}
 	return nil
 }
+
+func (r *Reconciler) deleteSpaceBindingRequest(logger logr.Logger, sbrAssociated *SpaceBindingRequestAssociated) error {
+	spaceBindingRequest := &toolchainv1alpha1.SpaceBindingRequest{}
+	memberClusterWithSpaceBindingRequest, found, err := cluster.LookupMember(r.MemberClusters, types.NamespacedName{
+		Namespace: sbrAssociated.namespace,
+		Name:      sbrAssociated.name,
+	}, spaceBindingRequest)
+	if err != nil {
+		if !found {
+			// got error while searching for SpaceBindingRequest CR
+			return err
+		}
+		// Just log the error but proceed because we did find the member anyway
+		logger.Error(err, "error while searching for SpaceBindingRequest")
+	} else if !found {
+		// let's just log the info
+		logger.Info("unable to find SpaceBindingRequest", "SpaceBindingRequest.Name", sbrAssociated.name, "SpaceBindingRequest.Namespace", sbrAssociated.namespace)
+		// try and delete the SpaceBinding since SBR was not found
+		return r.DeleteSpaceBinding(sbrAssociated.spaceBinding)
+	}
+
+	// delete the SBR
+	if err := memberClusterWithSpaceBindingRequest.Client.Delete(context.TODO(), spaceBindingRequest); err != nil {
+		return errs.Wrapf(err, "unable to delete the SpaceBindingRequest")
+	}
+	return nil
```

Comment:
I would also schedule another reconcile in let's say 10 seconds, to be sure that the controller would get back to the SpaceBinding again - to be sure that the deletion was completed.

Reply:
+1

Reply:
Good point, done in https://github.com/codeready-toolchain/host-operator/pull/857/commits/9b0996d46106b741e974909d6f023e48f7e6b588

Thanks for your suggestions!